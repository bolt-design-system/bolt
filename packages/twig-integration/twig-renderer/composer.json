{
  "name": "bolt-design-system/twig-renderer",
  "type": "library",
  "description": "Twig rendering service used by the Bolt Design System.",
  "version": "2.20.0",
  "license": "MIT",
  "repositories": [
    {
      "type": "path",
	    "url": "../twig-extensions-shared"
	  },
    {
      "type": "path",
      "url": "../twig-extensions-compat"
    }
  ],
  "require": {
    "bolt-design-system/core-php": "*",
    "bolt-design-system/drupal-twig-extensions": "*",
    "twig/twig": "^1.42.5",
    "webmozart/path-util": "*",
    "symfony/finder": "*"
  },
  "scripts": {
    "setup": "@composer install --no-interaction --prefer-dist --no-progress"
  }
}
