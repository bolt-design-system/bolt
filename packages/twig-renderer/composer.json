{
  "name": "bolt-design-system/twig-renderer",
  "type": "library",
  "version": "2.3.0",
  "license": "MIT",
  "require": {
    "bolt-design-system/core-php": "^2.3.0",
    "bolt-design-system/drupal-twig-extensions": "*",
    "twig/twig": "^1.38.2",
    "webmozart/path-util": "^2.3",
    "symfony/finder": "*"
  },
  "scripts": {
    "setup": "@composer install --no-interaction --prefer-dist --no-progress"
  }
}
