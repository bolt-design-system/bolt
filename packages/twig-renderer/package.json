{
  "name": "@bolt/twig-renderer",
  "version": "2.7.0",
  "license": "MIT",
  "description": "Renders Twig templates using an on-the-fly PHP rendering service.",
  "main": "twig-renderer.js",
  "repository": "https://github.com/bolt-design-system/bolt/tree/master/packages/twig-renderer",
  "keywords": [
    "design system",
    "bolt design system",
    "twig",
    "php"
  ],
  "maintainers": [
    {
      "name": "Salem Ghoweri",
      "email": "me@salemghoweri.com",
      "web": "https://github.com/sghoweri"
    }
  ],
  "dependencies": {
    "@basalt/twig-renderer": "^0.12.1",
    "@bolt/build-utils": "^2.7.0",
    "@bolt/drupal-twig-extensions": "^2.7.0",
    "sleep-promise": "^8.0.1"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "setup": "composer run setup"
  },
  "schema": "utils/config.schema.yml",
  "bugs": "https://github.com/bolt-design-system/bolt/issues",
  "homepage": "https://boltdesignsystem.com",
  "gitHead": "3d198feea5f73c63913a956a2a077e229ad8d462"
}
