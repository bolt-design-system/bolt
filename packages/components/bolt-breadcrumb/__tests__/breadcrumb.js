import { render, renderString } from '@bolt/twig-renderer';

describe('<bolt-breadcrumb> Component', () => {
  test('basic usage with attributes', async () => {
    const linkOne = await render('@bolt-components-link/link.twig', {
      text: 'Home',
      url: '#!',
    });
    const linkTwo = await render('@bolt-components-link/link.twig', {
      text: 'Other Page',
      url: '#!',
    });
    const results = await render(
      '@bolt-components-breadcrumb/breadcrumb.twig',
      {
        contentItems: [linkOne.html, linkTwo.html],
        attributes: {
          'data-foobar': 'baz',
          'aria-role': 'list',
        },
      },
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('basic usage with contentItems including rendered components and strings', async () => {
    const results = await renderString(`
{% include "@bolt-components-breadcrumb/breadcrumb.twig" with {
  contentItems: [
    include("@bolt-components-link/link.twig", {
      text: "Home",
      url: "#!"
    }),
    include("@bolt-components-link/link.twig", {
      text: "Landing Page",
      url: "#!"
    }),
    include("@bolt-components-link/link.twig", {
      text: "Sub Page",
      url: "#!"
    }),
    "Current Page"
  ]
} only %}
    `);
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('current page aria variation', async () => {
    const results = await renderString(`
{% include "@bolt-components-breadcrumb/breadcrumb.twig" with {
  contentItems: [
    include("@bolt-components-link/link.twig", {
      text: "Home",
      url: "#!"
    }),
    include("@bolt-components-link/link.twig", {
      text: "Landing Page",
      url: "#!"
    }),
    include("@bolt-components-link/link.twig", {
      text: "Sub Page",
      url: "#!"
    }),
    include("@bolt-components-link/link.twig", {
      text: "Current Page",
      url: "#!",
      attributes: {
        "aria-current": true
      },
    }),
  ]
} only %}
    `);
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
