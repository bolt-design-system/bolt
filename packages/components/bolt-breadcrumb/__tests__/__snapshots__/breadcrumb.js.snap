// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt Breadcrumb basic usage with contentItems including rendered components and strings 1`] = `
<bolt-breadcrumb bolt-component>
  <nav class="c-bolt-breadcrumb"
       aria-label="Breadcrumb"
       role="navigation"
  >
    <ol class="c-bolt-breadcrumb__list">
      <li class="c-bolt-breadcrumb__item">
        <bolt-link url="#!">
          <a href="#!"
             target="_self"
             class="c-bolt-link c-bolt-link--display-inline c-bolt-link--valign-center"
          >
            <ssr-keep for="bolt-link"
                      class="c-bolt-link__text"
            >
              Home
            </ssr-keep>
          </a>
        </bolt-link>
      </li>
      <li class="c-bolt-breadcrumb__item">
        <bolt-link url="#!">
          <a href="#!"
             target="_self"
             class="c-bolt-link c-bolt-link--display-inline c-bolt-link--valign-center"
          >
            <ssr-keep for="bolt-link"
                      class="c-bolt-link__text"
            >
              Other Page
            </ssr-keep>
          </a>
        </bolt-link>
      </li>
      <li class="c-bolt-breadcrumb__item">
        <bolt-link url="#!">
          <a href="#!"
             target="_self"
             class="c-bolt-link c-bolt-link--display-inline c-bolt-link--valign-center"
          >
            <ssr-keep for="bolt-link"
                      class="c-bolt-link__text"
            >
              Sub Page
            </ssr-keep>
          </a>
        </bolt-link>
      </li>
      <li class="c-bolt-breadcrumb__item">
        <bolt-link aria-current
                   url="#!"
        >
          <a href="#!"
             target="_self"
             class="c-bolt-link c-bolt-link--display-inline c-bolt-link--valign-center"
          >
            <ssr-keep for="bolt-link"
                      class="c-bolt-link__text"
            >
              Fourth Page
            </ssr-keep>
          </a>
        </bolt-link>
      </li>
    </ol>
  </nav>
</bolt-breadcrumb>
`;

exports[`Bolt Breadcrumb default 1`] = `
<bolt-breadcrumb bolt-component>
  <nav class="c-bolt-breadcrumb"
       aria-label="Breadcrumb"
       role="navigation"
  >
    <ol class="c-bolt-breadcrumb__list">
      <li class="c-bolt-breadcrumb__item">
        <bolt-link url="#!">
          <a href="#!"
             target="_self"
             class="c-bolt-link c-bolt-link--display-inline c-bolt-link--valign-center"
          >
            <ssr-keep for="bolt-link"
                      class="c-bolt-link__text"
            >
              Home
            </ssr-keep>
          </a>
        </bolt-link>
      </li>
      <li class="c-bolt-breadcrumb__item">
        <bolt-link url="#!">
          <a href="#!"
             target="_self"
             class="c-bolt-link c-bolt-link--display-inline c-bolt-link--valign-center"
          >
            <ssr-keep for="bolt-link"
                      class="c-bolt-link__text"
            >
              Other Page
            </ssr-keep>
          </a>
        </bolt-link>
      </li>
      <li class="c-bolt-breadcrumb__item">
        <bolt-link url="#!">
          <a href="#!"
             target="_self"
             class="c-bolt-link c-bolt-link--display-inline c-bolt-link--valign-center"
          >
            <ssr-keep for="bolt-link"
                      class="c-bolt-link__text"
            >
              Sub Page
            </ssr-keep>
          </a>
        </bolt-link>
      </li>
    </ol>
  </nav>
</bolt-breadcrumb>
`;
